lib/internal/streams/destroy.js

```diff
@@ -86,6 +88,14 @@ function _destroy(self, err, cb) {
     const r = self._readableState;
     const w = self._writableState;
 
+    if (!err) {
+      if (r?.autoDestroy && !isReadableFinished(self)) {
+        err = new AbortError();
+      } else if (w?.autoDestroy && !isWritableEnded(self)) {
+        err = new AbortError();
+      }
```

Comment:
```suggestion
      if ((r?.autoDestroy && !isReadableFinished(self)) ||
          (w?.autoDestroy && !isWritableEnded(self))) {
        err = new AbortError();
      }
```
This can be inlined.